rust_src/src/buffers.rs

```diff
@@ -23,42 +23,42 @@ pub type LispOverlayRef = ExternalPtr<Lisp_Overlay>;
 
 impl LispBufferRef {
     #[inline]
-    pub fn zv(&self) -> ptrdiff_t {
+    pub fn zv(self) -> ptrdiff_t {
         self.zv
     }
 
     #[inline]
-    pub fn pt(&self) -> ptrdiff_t {
+    pub fn pt(self) -> ptrdiff_t {
         self.pt
     }
 
     #[inline]
-    pub fn beg_addr(&self) -> *mut c_uchar {
+    pub fn beg_addr(&mut self) -> *mut c_uchar {
```

Comment:
This should also become `self`. (You might say that this discards the mutability requirement, but it carries an ownership requirement which is stronger.)

Reply:
I like that the mutable reference lets the developer know that the use of this method will change the internals of the object. I almost made a `beg_addr_const` method since not every use needs a mutable pointer as shown later in your review. To my eye we are being too sneaky since that mutable pointer should have a lifetime attached to it.

Reply:
Well, it makes no sense to use different conventions here. This

Reply:
In any case, it's a truth in the garbage-collected, single-threaded world of Emacs Lisp objects that there is no need for the "mutable XOR shared" principle that Rust references and lifetimes enforce.

Reply:
Fair enough. This is one of those places I think remacs can do better. But I see your point too.